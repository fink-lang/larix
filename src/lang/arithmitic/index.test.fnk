{describe, it, expect, to_match_snapshot} = import '../../testing/jest'

{parse_expr} = import '../../'


describe:: 'arithmitic', fn:

  it:: 'parses single line: a + b * c', fn:
    expect::
      parse_expr:: `a + b * c`
      to_match_snapshot


  it:: 'parses single line: a * b + c', fn:
    expect::
      parse_expr:: `a * b + c`
      to_match_snapshot


  it:: 'parses single line: a % b / c', fn:
    expect::
      parse_expr:: `a % b / c`
      to_match_snapshot


  it:: 'parses single line: a ^ b * c', fn:
    expect::
      parse_expr:: `a ^ b * c`
      to_match_snapshot


  it:: 'parses single line: a * -b', fn:
    expect::
      parse_expr:: `a * -b`
      to_match_snapshot


  it:: 'parses grouped: (a - b) * c', fn:
    expect::
      parse_expr:: `(a - b) * c`
      to_match_snapshot


  it:: 'parses multiline', fn:
    expect::
      parse_expr:: `
        a
          + b +
          c
            + d
      `
      to_match_snapshot

