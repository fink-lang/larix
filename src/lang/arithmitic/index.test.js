import {parse_expr} from '../../';


describe('arithmitic', ()=> {

  it('parses single line: a + b * c', ()=> {
    expect(
      parse_expr(`a + b * c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a * b + c', ()=> {
    expect(
      parse_expr(`a * b + c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a % b / c', ()=> {
    expect(
      parse_expr(`a % b / c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a ^ b * c', ()=> {
    expect(
      parse_expr(`a ^ b * c`)
    ).toMatchSnapshot();
  });


  it('parses single line: a * -b', ()=> {
    expect(
      parse_expr(`a * -b`)
    ).toMatchSnapshot();
  });


  it('parses grouped: (a - b) * c', ()=> {
    expect(
      parse_expr(`(a - b) * c`)
    ).toMatchSnapshot();
  });
});
