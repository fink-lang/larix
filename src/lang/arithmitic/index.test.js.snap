// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arithmitic parses grouped: (a - b) * c 1`] = `
arithm * (1:0-1:11)
  group (1:0-1:7)
    arithm - (1:1-1:6)
      other (1:1-1:2) a
      other (1:5-1:6) b
  other (1:10-1:11) c
`;

exports[`arithmitic parses single line: a % b / c 1`] = `
arithm / (1:0-1:9)
  arithm % (1:0-1:5)
    other (1:0-1:1) a
    other (1:4-1:5) b
  other (1:8-1:9) c
`;

exports[`arithmitic parses single line: a * -b 1`] = `
arithm * (1:0-1:6)
  other (1:0-1:1) a
  arithm_prefix - (1:4-1:6)
    other (1:5-1:6) b
`;

exports[`arithmitic parses single line: a * b + c 1`] = `
arithm + (1:0-1:9)
  arithm * (1:0-1:5)
    other (1:0-1:1) a
    other (1:4-1:5) b
  other (1:8-1:9) c
`;

exports[`arithmitic parses single line: a + b * c 1`] = `
arithm + (1:0-1:9)
  other (1:0-1:1) a
  arithm * (1:4-1:9)
    other (1:4-1:5) b
    other (1:8-1:9) c
`;

exports[`arithmitic parses single line: a ^ b * c 1`] = `
arithm * (1:0-1:9)
  arithm_right ^ (1:0-1:5)
    other (1:0-1:1) a
    other (1:4-1:5) b
  other (1:8-1:9) c
`;
